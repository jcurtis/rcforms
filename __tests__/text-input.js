import React from 'react'
import renderer from 'react-test-renderer'
import Form, {TextInput} from '../lib/index.js'

test('TextInput renders', () => {
  const component = renderer.create(
    <TextInput />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('TextInput renders with props', () => {
  const component = renderer.create(
    <TextInput
      name='name'
      readOnly
    />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('TextInput renders inside a Form', () => {
  const component = renderer.create(
    <Form>
      <TextInput />
    </Form>
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('TextInput is composable', () => {
  function ComposedInput ({label}) {
    return (
      <div className='composed'>
        <span>{label}</span>
        <TextInput />
      </div>
    )
  }
  const component = renderer.create(
    <Form>
      <h1>My Form</h1>
      <ComposedInput label='Composable' />
    </Form>
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
