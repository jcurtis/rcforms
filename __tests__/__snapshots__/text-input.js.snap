// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput is composable 1`] = `
<form
  className="form"
  noValidate={true}
  role="form"
>
  <h1>
    My Form
  </h1>
  <div
    className="composed"
  >
    <span>
      Composable
    </span>
    <input
      onBlur={[Function]}
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
</form>
`;

exports[`TextInput renders 1`] = `
<input
  onBlur={[Function]}
  onChange={[Function]}
  type="text"
  value=""
/>
`;

exports[`TextInput renders inside a Form 1`] = `
<form
  className="form"
  noValidate={true}
  role="form"
>
  <input
    onBlur={[Function]}
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;

exports[`TextInput renders with props 1`] = `
<input
  name="name"
  onBlur={[Function]}
  onChange={[Function]}
  readOnly={true}
  type="text"
  value=""
/>
`;
